internal/actuator/actuator.go

```diff
@@ -25,65 +25,62 @@ func NewActuator(k8sClient client.Client) *Actuator {
 	}
 }
 
-func (a *Actuator) ApplyReplicaTargets(ctx context.Context, VariantAutoscaling *llmdOptv1alpha1.VariantAutoscaling) error {
-	desired := VariantAutoscaling.Status.DesiredOptimizedAlloc
+// getCurrentDeploymentReplicas gets the real current replica count from the actual Deployment
+func (a *Actuator) getCurrentDeploymentReplicas(ctx context.Context, va *llmdOptv1alpha1.VariantAutoscaling) (int32, error) {
 	var deploy appsv1.Deployment
 	err := a.Client.Get(ctx, types.NamespacedName{
-		Name:      VariantAutoscaling.Name,
-		Namespace: VariantAutoscaling.Namespace,
+		Name:      va.Name,
+		Namespace: va.Namespace,
 	}, &deploy)
 	if err != nil {
-		return fmt.Errorf("failed to get Deployment %s/%s: %w", VariantAutoscaling.Namespace, VariantAutoscaling.Name, err)
+		return 0, fmt.Errorf("failed to get Deployment %s/%s: %w", va.Namespace, va.Name, err)
 	}
 
-	// Patch replicas field
-	original := deploy.DeepCopy()
-	replicas := int32(desired.NumReplicas)
-	deploy.Spec.Replicas = &replicas
-
-	patch := client.MergeFrom(original)
-	if err := a.Client.Patch(ctx, &deploy, patch); err != nil {
-		return fmt.Errorf("failed to patch Deployment %s: %w", deploy.Name, err)
+	// Prefer status replicas (actual current state)
+	if deploy.Status.Replicas > 0 {
+		return deploy.Status.Replicas, nil
 	}
 
-	logger.Log.Info("Patched Deployment: ", "name: ", deploy.Name, ", num-replicas: ", replicas)
-
-	// Emit metrics for replica scaling
-	if replicas > *original.Spec.Replicas {
-		if err := a.MetricsEmitter.EmitReplicaScalingMetrics(ctx, VariantAutoscaling, "scale_up", "load_increase"); err != nil {
-			logger.Log.Error(err, "Failed to emit scale-up metrics for variantAutoscaling - ", "variantAutoscaling-name: ", VariantAutoscaling.Name)
-			// Don't fail the deployment patch for metric emission errors
-		}
-	} else if replicas < *original.Spec.Replicas {
-		if err := a.MetricsEmitter.EmitReplicaScalingMetrics(ctx, VariantAutoscaling, "scale_down", "load_decrease"); err != nil {
-			logger.Log.Error(err, "Failed to emit scale-down metrics for variantAutoscaling - ", "variantAutoscaling-name: ", VariantAutoscaling.Name)
-			// Don't fail the deployment patch for metric emission errors
-		}
+	// Fallback to spec if status not ready
+	if deploy.Spec.Replicas != nil {
+		return *deploy.Spec.Replicas, nil
 	}
 
-	return nil
+	// Final fallback
+	return 1, nil
```

Comment:
Final fallback should not be 1, it looks like we don't need this function anymore.

Reply:
> That fallback is an edge case where deployment exists but has no `spec.replicas` set

spec.replicas defaults to 1 as per the deployment controller design.